core/app/c/[communitySlug]/pubs/PubList.tsx

```diff
@@ -94,23 +120,22 @@ const PubListFooterPagination = async (props: {
 	communityId: CommunitiesId;
 	children: React.ReactNode;
 	pubsPromise: Promise<ProcessedPub[]>;
+	userId: UsersId;
 }) => {
 	const perPage = searchParamsCache.get("perPage");
-	const isQuery = !!searchParamsCache.get("query");
-
-	const count = await (isQuery
-		? props.pubsPromise.then((pubs) => pubs.length)
-		: getPubsCount({ communityId: props.communityId }));
-
-	const paginationProps = isQuery
-		? {
-				mode: "cursor" as const,
-				hasNextPage: count > perPage,
-			}
-		: {
-				mode: "total" as const,
-				totalPages: Math.ceil((count ?? 0) / perPage),
-			};
+	const query = searchParamsCache.get("query");
+
+	const count = await getPubsCount(
+		{ communityId: props.communityId, userId: props.userId },
+		{
+			search: query,
+		}
+	);
+
+	const paginationProps = {
+		mode: "total" as const,
+		totalPages: Math.ceil((count ?? 0) / perPage),
+	};
```

Comment:
i initially added some different type of pagination for search queries bc we couldnt count the total number of results. we can now, so we always do the normal pagination